src/app/components/message/MsgTypeRenderers.tsx

```diff
@@ -80,6 +88,22 @@ export function MText({ edited, content, renderBody, renderUrlsPreview }: MTextP
   const trimmedBody = trimReplyFromBody(body);
   const urlsMatch = renderUrlsPreview && trimmedBody.match(URL_REG);
   const urls = urlsMatch ? [...new Set(urlsMatch)] : undefined;
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const { shouldTruncate, finalContent, finalCustomContent } = useMemo(() => {
+    const contentStr = trimmedBody || '';
+    const customContentStr = typeof customBody === 'string' ? trimReplyFromBody(customBody) : '';
+    const needTruncate = contentStr.length > 750 || customContentStr.length > 750;
+    return {
+      contentStr,
+      customContentStr,
+      shouldTruncate: needTruncate,
+      finalContent: isExpanded ? contentStr : truncateText(contentStr, CHARACTER_LIMIT),
+      finalCustomContent: isExpanded
+        ? customContentStr
+        : truncateText(customContentStr, CHARACTER_LIMIT),
+    };
+  }, [trimmedBody, customBody, isExpanded]);
```

Comment:
I have following concern:
1. using the memoization is not really needed here as we are not doing any time consuming computation and conditional logic can be used directly while rendering.
2. `contentStr` and `customContentStr` are just the copy of `body` and `customBody` without any changes.
3. `trimReplyFromBody` is used on `customBody` and it is only meant to be used on plain `body`.
4. `truncateText` cut the customHTML body right after 750 chars which can breaks the customHTML tags entirely.

Reply:
Done . Now truncation of customHTML tags is done correctly.